aviatrix/resource_aviatrix_smart_group.go

```diff
@@ -155,39 +176,32 @@ func marshalSmartGroupInput(d *schema.ResourceData) (*goaviatrix.SmartGroup, err
 		selectorInfo := selectorInterface.(map[string]interface{})
 		var filter *goaviatrix.SmartGroupMatchExpression
 
-		if mapContains(selectorInfo, "cidr") || mapContains(selectorInfo, "fqdn") || mapContains(selectorInfo, "site") {
-			for _, key := range []string{"type", "res_id", "account_id", "account_name", "name", "region", "zone", "tags"} {
-				if mapContains(selectorInfo, key) {
-					return nil, fmt.Errorf("%q must be empty when %q is set", key, "cidr")
-				}
+		// Handle the case of the non-tag selector
+		if matchingKey, ok := goaviatrix.MapContainsOneOfKeys(selectorInfo, nonTagKeys); ok {
+			// Error if both the tagged and non-tagged selectors are present together
+			if matchingType, okType := goaviatrix.MapContainsOneOfKeys(selectorInfo, tagKeys); okType {
+				return nil, fmt.Errorf("%q must be empty when %q is set", matchingType, matchingKey)
```

Comment:
```suggestion
		// Handle the case of the non-tag selector
		if matchingKey, ok := goaviatrix.MapContainsOneOfKeys(selectorInfo, exclusiveFilterKeys); ok {
			// Error if both an exclusive and non-exclusive filter selector are present together
			if matchingType, okType := goaviatrix.MapContainsOneOfKeys(selectorInfo, filterKeys); okType {
				return nil, fmt.Errorf("%q must be empty when %q is set", matchingType, matchingKey)
```

Do you also need a condition here for if two exclusive keys are used? E.g. `cidr` and `fqdn`?